mindsdb/interfaces/knowledge_base/controller.py

```diff
@@ -123,11 +123,10 @@ def insert(self, df: pd.DataFrame):
 
         # add embeddings
         df_emb = self._df_to_embeddings(df)
-        df = pd.concat([df, df_emb], axis=1)
 
         # send to vector db
         db_handler = self._get_vector_db()
-        db_handler.do_upsert(self._kb.vector_database_table, df)
+        db_handler.do_upsert(self._kb.vector_database_table, df_emb)
```

Comment:
df_emb doesn't content text ('content' value). how it works for you?

Reply:
It works on my system.. is there an error it is throwing for you?

Reply:
Discussed yesterday with @tmichaeldb 
Some embedding models returns input column (cohere) and some not (openai)
- _df_to_embeddings should filter model output and return only one column (embeddings)
- return concat of  df and df_emb